### PR TITLE
feat support for tgwa peering and accepter

### PR DESCRIPTION
**Pull request by arkhoss**
## Description
Feat: support for transit gateway attachment type peering and its accepter

## Motivation and Context
Need to support multi-region scenarios with tgw peering to tgw

## Breaking Changes
none

## How Has This Been Tested?
Run over tf and tg plan and apply with no infra changes.

### CODE REVIEW

**Review by adamwshero — CHANGES_REQUESTED** [review]
Looks good @arkhoss . 

1. Please update Terragrunt and Terraform examples. 
2. Also update the CHANGELOG as a minor version update and README where applicable.
3. Create a release for the new tagged version once merged and tags are pushed

File: README.md [hunk 1]
```diff
@@ -122,46 +122,53 @@ inputs = {
...
+| [aws_ec2_transit_gateway_peering_attachment_accepter.this](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_transit_gateway_peering_attachment_accepter) | resource |
+| [aws_ec2_transit_gateway_vpc_attachment.this](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_transit_gateway_vpc_attachment) | resource |
+| [aws_route.this_local_gateway_route](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route) | resource |
+| [aws_route.this_nat_gateway_route](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route) | resource |
+| [aws_route.this_network_interface_route](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route) | resource |
+| [aws_route.this_tgw_route](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route) | resource |
+| [aws_route.this_vpc_endpoint_route](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route) | resource |
+| [aws_route.this_vpc_peering_route](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route) | resource |
+
```
Comment (adamwshero) [inline README.md:142]: This list above needs to include the new functionality being added

File: README.md
Comment (adamwshero) [inline README.md:1]: - Line 17, need to include new functionality that has been added.
- Does anything need to be mentioned under "Assumptions"
- Same question for usage
@arkhoss

Comment (arkhoss) [reply README.md:1]: I think all as requested was resolved, please take a look @adamwshero

**Review by adamwshero — APPROVED** [review]

**Pull request merged**